### PR TITLE
Responsive replay

### PR DESCRIPTION
**Pull request by SebastienTainon**
The goal here is to make it possible to force a zoom level to every visualization (editor, variables view, 1d array, 2d array, etc...) so that when the teacher creates a recording he has a high zoom level and must scroll to see things, and when the student sees the recording he has a low zoom level to see as many things as possible on the screen. As a bonus, the teacher can himself choose to record with a mobile/tablet layout to be sure he is in the "worst" condition.

### CODE REVIEW

**Review by mblockelet — COMMENTED** [review]
Just simple comments on the interface text, however I can't fully test as it's giving me an error in recording mode :
```
Source: recorderPrepare

TypeError: Cannot read properties of undefined (reading 'call')
```

File: frontend/lang/en-US.js [hunk 1]
```diff
@@ -153,4 +154,6 @@ module.exports = {
     TASK_EDITOR: 'Editor',
     TASK_IO: 'Terminal',
     TASK_VISUALIZATION: "Visualization {number}",
+
+    LAYOUT_TYPE_REQUIRED_LABEL: "Force a layout:",
```
[the remark this answers is not in the document]
Comment (SebastienTainon) [reply frontend/lang/en-US.js:158]: That's technically not a preview.... The teacher can force himself to be in a configuration where he sees the least number of visualization panes at a time, and is thus forced to navigate from one visualization to another.

File: frontend/common/SettingsDialog.tsx [hunk 2]
```diff
@@ -43,11 +46,27 @@ export function SettingsDialog(props: SettingsDialogProps) {
...
+        const requiredLayout = event.target.value;
+        dispatch({type: LayoutActionTypes.LayoutRequiredTypeChanged, payload: {requiredType: requiredLayout ? requiredLayout : null}});
+    };
+
     const modeOptions = [
         {value: IoMode.Split, label: 'IOPANE_MODE_SPLIT'},
         {value: IoMode.Terminal, label: 'IOPANE_MODE_INTERACTIVE'}
     ];
 
+    const layoutChoices = [
```
[the remark this answers is not in the document]
Comment (SebastienTainon) [reply frontend/common/SettingsDialog.tsx:59]: Good idea, I've just committed this :+1:

**Comment by SebastienTainon** [comment]
> Just simple comments on the interface text, however I can't fully test as it's giving me an error in recording mode :
> 
> ```
> Source: recorderPrepare
> 
> TypeError: Cannot read properties of undefined (reading 'call')
> ```

That was caused by HTTP. It should work in HTTPS ;)

**Review by mblockelet — APPROVED** [review]
:+1:

**Pull request merged**